aten/src/ATen/native/mkldnn/Utils.cpp

```diff
@@ -28,4 +29,30 @@ std::vector<int64_t> conv_output_size(
   return output_size;
 }
 
+std::vector<int64_t> pool_output_sizes(
+    IntArrayRef input_size,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode) {
+  std::vector<int64_t> output_size(input_size.size());
+  // copy N and C
+  output_size[0] = input_size[0];
+  output_size[1] = input_size[1];
+
+  for (int i = 2; i < input_size.size(); ++i) {
+    output_size[i] = pooling_output_shape<int64_t>(
+      input_size[i],
+      kernel_size[i - 2],
+      padding[i - 2],
+      stride[i - 2],
+      dilation[i - 2],
+      ceil_mode
+    );
+  }
+
```

Comment:
MKLDNN treats padding differently with ceil mode.
e.g. the following case would result runtime error, both max pooling and avg pooling:
```python
import torch
x = torch.randn(1, 1, 10, 10, dtype=torch.float32) * 10
max_pool2d = torch.nn.MaxPool2d(kernel_size=(4,5),stride=(3,4),padding=(1,2), ceil_mode=True)
# max_pool2d = torch.nn.AvgPool2d(kernel_size=(4,5),stride=(3,4),padding=(1,2), ceil_mode=True)
output =  max_pool2d(x.to_mkldnn()).to_dense()
```
please refer [caffe ](https://github.com/intel/caffe/blob/master/src/caffe/layers/mkldnn_pooling_layer.cpp#L122) code of computing pooling output size for detail. @XiaobingSuper

Reply:
has this comment been addressed? (sorry I didn't follow the point)

Reply:
This problem can be solved by change the mkldnn padding as caffe's doing: https://github.com/intel/caffe/blob/362a3b375e8f5bc64f5338d18f1742448cfb8fb7/src/caffe/layers/mkldnn_pooling_layer.cpp#L148

Reply:
Please note that mkldnn is to get given output size by changing padding size.

Reply:
Added an assert to only support non ceil_mode for now. Let's do the padding trick in a follow up PR.